samples/openvx/wrappers.cpp

```diff
@@ -0,0 +1,214 @@
+    Node::create(graph, VX_KERNEL_THRESHOLD, mag8, threshMag, thrm);
+    Node::create(graph, VX_KERNEL_AND, thra, thrm, thrv);
+    Node::create(graph, VX_KERNEL_AND, inputImage, thrv, outputImage);
+
+    graph.verify();
+
+    return graph;
+}
+
+
+int ovxDemo(std::string inputPath, bool useCopy, bool mapFromVx)
+{
+    using namespace cv;
+    using namespace ivx;
+
+    Mat image = imread(inputPath, IMREAD_GRAYSCALE);
+    if (image.empty()) return -1;
+
+    //check image format
+    if (image.depth() != CV_8U || image.channels() != 1) return -1;
+
+    try
+    {
+        Context context = Context::create();
+        //put user data from cv::Mat to vx_image
+        vx_df_image color = Image::matTypeToFormat(image.type());
+        vx_uint32 width = image.cols, height = image.rows;
+        Image ivxImage;
+        if (useCopy)
+        {
+            ivxImage = Image::create(context, width, height, color);
+            ivxImage.copyFrom(0, image);
+        }
+        else
+        {
+            vx_imagepatch_addressing_t addressing = Image::createAddressing(width, height, (vx_uint32)image.elemSize(),
+                                                                            (vx_uint32)image.step);
+            const std::vector<vx_imagepatch_addressing_t> addrs(1, addressing);
+            const std::vector<void*> ptrs(1, image.data);
+            ivxImage = Image::createFromHandle(context, color, addrs, ptrs);
+        }
+
+        Image ivxResult;
+        Image::Patch resultPatch;
+        Mat output;
+        if (useCopy || mapFromVx)
+        {
+            //we will copy or map data from vx_image to cv::Mat
+            ivxResult = ivx::Image::create(context, width, height, VX_DF_IMAGE_U8);
+        }
+        else if (!mapFromVx)
+        {
+            //create vx_image based on user data, no copying required
+            output = cv::Mat(height, width, CV_8U, cv::Scalar(0));
+            vx_imagepatch_addressing_t addressing = Image::createAddressing(width, height, (vx_uint32)output.elemSize(),
+                                                                            (vx_uint32)output.step);
+            const std::vector<vx_imagepatch_addressing_t> addrs(1, addressing);
+            const std::vector<void*> ptrs(1, output.data);
+            ivxResult = Image::createFromHandle(context, Image::matTypeToFormat(CV_8U), addrs, ptrs);
+        }
+
+        Graph graph = createProcessingGraph(context, ivxImage, ivxResult);
+
+        // Graph execution
+        graph.process();
+
+        //getting resulting image in cv::Mat
+        if (useCopy)
+        {
+            ivxResult.copyTo(0, output);
+        }
+        else
+        {
+            if (mapFromVx)
+            {
+                //create cv::Mat based on vx_image mapped data
+                resultPatch.map(ivxResult, 0, ivxResult.getValidRegion(), VX_READ_AND_WRITE);
+                output = resultPatch.getMat();
```

Comment:
this code is dangerous in a general case!
I suggest adding a comment here:
`// generally this is very bad idea! but in our case unmap() wan't happen until output is in use`